tests/github-glue.test.ts

```diff
@@ -0,0 +1,204 @@
+import { Octokit } from "@octokit/rest";
+import "jest";
+import { git, gitConfig } from "../lib/git";
+import { GitHubGlue } from "../lib/github-glue";
+
+/*
+This test requires setup.  It will run successfully if setup has
+not been done.  If the test fails, there may be a pull request and a
+branch to be deleted on github and the local repo.  The test will
+attempt cleanup of a previously failed test.
+
+Setup:
+gitgitgadget.githubtest.githubuser must be configured for this test to
+identify your GitHub login.
+The value must also be configured to identify a test repo to use as
+gitgitgadget.<login>.githubrepo.
+
+Additionally, a GitHub personal access token must be set for the
+login to succeed.  This can be restricted to the test repo.
+
+For example, these configuration settings are needed (where
+`octokity` is your GitHub login):
+git config --add gitgitgadget.githubtest.githubuser octokity
+git config --add gitgitgadget.octokity.githubrepo ggg-test
+git config --add gitgitgadget.octokity.githubtoken feedbeef...
+
+The test repo must exist in github and locally.  It is expected to be
+located at the same directory level as this project.
+*/
+
+class GitHubProxy extends GitHubGlue {
+    public octo = new Octokit();
+    public constructor(workDir?: string, repo = "git") {
+        super(workDir, repo);
+        this.octo = this.client;
+    }
+
+    public async authenticate(repositoryOwner: string): Promise<void> {
+        await this.ensureAuthenticated(repositoryOwner);
+        this.octo = this.client;
+    }
+}
+
+jest.setTimeout(180000);
+
+test("identify user", async () => {
+    const owner = await gitConfig(`gitgitgadget.githubtest.githubuser`) || "";
+    const repo = await gitConfig(`gitgitgadget.${owner}.githubrepo`) || "";
```

Comment:
Maybe this could be done as part of a setup phase (see https://jestjs.io/docs/en/setup-teardown.html)?

Reply:
I can see using a common function to return this info in the test cases but not clear how a setup phase can provide owner and repo vars.  Not joking to say I am still learning jest.

Reply:
What I would do is to declare both via `let`, outside the functions, and then set then in the `beforeEach()` function.

Reply:
Changing to set with `beforeAll()`.  They only need to be set once.

Reply:
Excellent!